src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -37,227 +114,2408 @@ const ActorInit En_Ossan_InitVars = {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0:
+                    EnOssan_StartShopping(globalCtx, this);
+                    break;
+                case 1:
+                    EnOssan_EndInteraction(globalCtx, this);
+                    break;
+            }
+        }
+    } else if (dialogState == 5 && func_80106BC8(globalCtx)) {
+        func_80078884(NA_SE_SY_MESSAGE_PASS);
+
+        switch (this->happyMaskShopState) {
+            case OSSAN_HAPPY_STATE_ALL_MASKS_SOLD:
+                func_8010B720(globalCtx, 0x70AA);
+                this->stateFlag = OSSAN_STATE_LEND_MASK_OF_TRUTH;
+                return;
+            case OSSAN_HAPPY_STATE_BORROWED_FIRST_MASK:
+                EnOssan_EndInteraction(globalCtx, this);
+                return;
+            case OSSAN_HAPPY_STATE_REQUEST_PAYMENT_KEATON_MASK:
+            case OSSAN_HAPPY_STATE_REQUEST_PAYMENT_SPOOKY_MASK:
+            case OSSAN_HAPPY_STATE_REQUEST_PAYMENT_SKULL_MASK:
+            case OSSAN_HAPPY_STATE_REQUEST_PAYMENT_BUNNY_HOOD:
+                EnOssan_TryPaybackMask(this, globalCtx);
+                return;
+            case OSSAN_HAPPY_STATE_ANGRY:
+                globalCtx->nextEntranceIndex = 0x1D1;
+                globalCtx->sceneLoadFlag = 0x14;
+                globalCtx->fadeTransition = 0x2E;
+                return;
+        }
+
+        if (!EnOssan_TestEndInteraction(this, globalCtx, &globalCtx->state.input[0])) {
+            // "Shop around by moving the stick left and right"
+            osSyncPrintf("「スティック左右で品物みてくれ！」\n");
+            EnOssan_StartShopping(globalCtx, this);
+        }
+    }
+
+    if (1) {}
+}
+
+s32 EnOssan_FacingShopkeeperDialogResult(EnOssan* this, GlobalContext* globalCtx) {
+    switch (globalCtx->msgCtx.choiceIndex) {
+        case 0:
+            EnOssan_ChooseTalkToOwner(globalCtx, this);
+            return true;
+        case 1:
+            EnOssan_EndInteraction(globalCtx, this);
+            return true;
+        default:
+            return false;
+    }
+}
+
+void EnOssan_State_FacingShopkeeper(EnOssan* this, GlobalContext* globalCtx, Player* player) {
+    u8 nextIndex;
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) &&
+        !EnOssan_TestEndInteraction(this, globalCtx, &globalCtx->state.input[0])) {
+        if (func_80106BC8(globalCtx) && EnOssan_FacingShopkeeperDialogResult(this, globalCtx)) {
+            func_80078884(NA_SE_SY_DECIDE);
+            return;
+        }
+        // Stick Left
+        if (this->stickAccumX < 0) {
+            nextIndex = EnOssan_SetCursorIndexFromNeutral(this, 4);
+            if (nextIndex != CURSOR_INVALID) {
+                this->cursorIndex = nextIndex;
+                this->stateFlag = OSSAN_STATE_LOOK_SHELF_LEFT;
+                Interface_SetDoAction(globalCtx, 6);
+                this->stickRightPrompt.isEnabled = false;
```

Comment:
Noticed this while working over in MM, but I think you have the stick prompts named backwards. For example here you are turning towards the shelf on the left, so I would think the left stick prompt should be disabled and the right one enabled.